website/docs/10_going-deep/instrumentation.md

```diff
@@ -0,0 +1,138 @@
+  }
+}
+```
+
+You can then configure the tracer in your Booster application's configuration:
+
+```typescript
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { MyTracer } from './my-tracer'
+
+const myTracer = new MyTracer()
+const config: BoosterConfig = {
+// ...other configuration options...
+  trace: {
+    enableTraceNotification: true,
+    verbose: false,
+    onStart: await myTracer.onStart,
+    onEnd: await myTracer.onStart,
+  }
+}
+```
+
+In the configuration above, we've enabled trace notifications and specified our onStart and onEnd as the methods to use. Verbose disable will reduce the amount of information generated excluding the internal parameter in the response. 
+
+By default, all traceable actions are disabled, but you can enable all of them or selectively enable only specific actions using an array of TraceActionTypes.
+
+```typescript
+import { BoosterConfig, TraceActionTypes } from '@boostercloud/framework-types'
+import { MyTracer } from './my-tracer'
+
+const myTracer = new MyTracer()
+const config: BoosterConfig = {
+// ...other configuration options...
+  trace: {
+    enableTraceNotification: [TraceActionTypes.DISPATCH_EVENT, TraceActionTypes.MIGRATION_RUN],
+    verbose: false,
+    onStart: await myTracer.onStart,
+    onEnd: await myTracer.onStart,
+  }
+}
+```
+
+In this example, only DISPATCH_EVENT and MIGRATION_RUN actions will trigger trace notifications.
+
+### TraceActionTypes
+The TraceActionTypes enum defines all the traceable actions in Booster's core:
+
+```typescript
+export enum TraceActionTypes {
+  CUSTOM,
+  EVENT_HANDLERS_PROCESS,
+  HANDLE_EVENT,
+  DISPATCH_ENTITY_TO_EVENT_HANDLERS,
+  DISPATCH_EVENTS,
+  FETCH_ENTITY_SNAPSHOT,
+  STORE_SNAPSHOT,
+  LOAD_LATEST_SNAPSHOT,
+  LOAD_EVENT_STREAM_SINCE,
+  ENTITY_REDUCER,
+  READ_MODEL_FIND_BY_ID,
+  GRAPHQL_READ_MODEL_SEARCH,
+  READ_MODEL_SEARCH,
+  COMMAND_HANDLER,
+  MIGRATION_RUN,
+  GRAPHQL_DISPATCH,
+  GRAPHQL_RUN_OPERATION,
+  SCHEDULED_COMMAND_HANDLER,
+  DISPATCH_SUBSCRIBER_NOTIFIER,
+  READ_MODEL_SCHEMA_MIGRATOR_RUN,
+  SCHEMA_MIGRATOR_MIGRATE,
+}
+```
+
+### TraceParameters
+The TraceParameters interface defines the data that is passed to the tracer's onBefore and onAfter methods:
+
+```typescript
+export interface TraceParameters {
+  className: string
+  methodName: string
+  args: Array<unknown>
+  traceId: UUID
+  elapsedInvocationMillis?: number
+  internal: {
+    target: unknown
+    descriptor: PropertyDescriptor
+  }
+  description?: string
+}
+```
+
+className and methodName identify the function that is being traced.
+
+### Adding the Trace Decorator to Your Own Methods
+In addition to using the Trace Decorator to receive notifications when events occur in Booster's core, you can also use it to trace your own methods. To add the Trace Decorator to your own methods, simply add @Trace() before your method declaration.
```

Comment:
If we eventually allow the user to use their own action names, this would need to change